### PR TITLE
Hide appender on BlockPreviews via checking for null component in BlockListAppender

### PR DESCRIPTION
**Pull request by getdave**
Previously if a `null` component was passed to the `BlockList` component it would still render an empty `<div className="block-list-appender">` element. This because we didn't check for `null` before rendering.

This is useful because we sometimes want to disable the rendering of the Block appender UI. For example in block previews.

Required for https://github.com/WordPress/gutenberg/pull/16873

## How has this been tested?
* Insert quote Block with content
* Switch style of Block
* Inspect the Block Preview of the Style. See no `<div className="block-list-appender">` component.

## Types of changes
* Bug fix (non-breaking change which fixes an issue) 
* New feature (non-breaking change which adds functionality)

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate.

### CODE REVIEW

File: packages/block-editor/src/components/block-list-appender/index.js [hunk 1]
```diff
@@ -27,8 +27,10 @@ function BlockListAppender( {
 		return null;
 	}
 
-	// A render prop has been provided, use it to render the appender.
-	if ( CustomAppender ) {
+	// A render prop has been provided
+	// and it is not a component that return `null`
+	// then use it to render the appender.
+	if ( CustomAppender && ! isNull( CustomAppender ) ) {
```
Comment (youknowriad) [inline packages/block-editor/src/components/block-list-appender/index.js]: I guess `isNull` is not correct here? We need to check that it's a function that returns a falsy value right? A function is not null?

Comment (youknowriad) [reply packages/block-editor/src/components/block-list-appender/index.js]: A simpler approach might be to do `CustomAppender === false` ? what do you think?

Comment (getdave) [reply packages/block-editor/src/components/block-list-appender/index.js]: As in 

`if ( CustomAppender && isFunction( CustomAppender ) && ! isNull( CustomAppender() ) ) {`?

Comment (youknowriad) [reply packages/block-editor/src/components/block-list-appender/index.js]: yeah but the second approach seems simpler. It ensures that if you do `renderAppender={ false }` there's no appender rendered? (I'm ok with both you choose :P )

**Review by youknowriad — APPROVED** [review]
Pushed a tweak, merge if it works for you.

**Comment by getdave** [comment]
Thanks @youknowriad. Using `false` is much clearer than passing a `null` Component.

**Pull request merged**